backend/app.py

```diff
@@ -633,6 +640,77 @@ def deleteChatConversation(chat_id):
         logging.exception("[webbackend] exception in /delete-chat-conversation")
         return jsonify({"error": str(e)}), 500
 
+#get report by id argument
+@app.route("/api/reports/<report_id>", methods=["GET"])
+def getReport(report_id):
+    """
+    Endpoint to get a report by ID.
+    """
+    try:
+        report = get_report(report_id)
+        return jsonify(report), 200
+    except Exception as e:
+        logging.exception(f"Error retrieving report with id {report_id}")
+        return jsonify({"error": str(e)}), 404
+
+#get report by type argument
+@app.route("/api/reports", methods=["GET"])
+def getReportsType():
+    """
+    Endpoint to obtain reports by type.
+    """
+    report_type = request.args.get("type")  # Obtener el parámetro `type` de la query string
+    if not report_type:
+        return jsonify({"error": "The 'type' query parameter is required"}), 400
+
+    try:
+        reports = get_report_by_type(report_type)
+        return jsonify(reports), 200
+    except Exception as e:
+        logging.exception(f"Error retrieving reports with type {report_type}")
+        return jsonify({"error": str(e)}), 500
+
+#create report
+@app.route("/api/reports", methods=["POST"])
+def createReport():
+    """
+    Endpoint to create a new report.
+    """
+    try:
+        data = request.get_json()
+        new_report = create_report(data)
+        return jsonify(new_report), 200
+    except Exception as e:
+        logging.exception("Error creating report")
+        return jsonify({"error": str(e)}), 500
+
+#update report
+@app.route("/api/reports/<report_id>", methods=["PUT"])
+def updateReport(report_id):
+    """
+    Endpoint to update a report by ID.
+    """
+    try:
+        updated_data = request.get_json()
+        updated_report = update_report(report_id, updated_data)
+        return jsonify(updated_report), 200
+    except Exception as e:
+        logging.exception(f"Error updating report with id {report_id}")
+        return jsonify({"error": str(e)}), 500
+
+#delete report
+@app.route("/api/reports/<report_id>", methods=["DELETE"])
```

Comment:
@ramirezmorac2  related to:
HTTP Status Codes:

Consider using 204 No Content on successful deletion when no additional body content is needed, as it clearly indicates the resource is gone.
If a non-existent report_id is requested, returning a 404 Not Found would be more appropriate than a generic 500 Internal Server Error.
Error Handling & Validation:

Introduce specific exception handling. For example, if delete_report raises a ReportNotFoundError, return 404 Not Found.
For generic, unexpected errors, 500 Internal Server Error is fine, but consider returning a more user-friendly and generic error message.
Response Consistency:

When a resource is successfully deleted, some APIs prefer returning no content (204). If a message is needed, 200 OK can remain, but ensure consistency with the rest of the API design.